### ISSUE TITLE
Market data not working as it is meant to (I think)

### ISSUE THREAD

**Issue by Jugom** [EDDiscovery/EDDiscovery#3274]
So, I was revisiting the Market data panel after my questions in discord and was preparing a little something to follow up. But it became so extensive that I figured I could just post it in here directly for you to review later. 

The sell only tick seems to be filtering fine. 

even when a commodity has 0 demand. although it doesn’t show it as 0 but just nukes the numeric value.
![image](https://user-images.githubusercontent.com/74990247/171162395-3acdbc95-6adb-4868-8198-8b25756bda3f.png)
![image](https://user-images.githubusercontent.com/74990247/171162432-c589a6dd-5a8f-4866-bbcb-e1b7e079ea89.png)

however, the Buy only tick, I don’t think is working how it is supposed to. 
if you click on it, it does hide all the commodities which have a demand, as they are from the sell category.
![image](https://user-images.githubusercontent.com/74990247/171162836-05c3dad7-4d75-42b3-9ac1-583b466744d0.png)
![image](https://user-images.githubusercontent.com/74990247/171162875-29f1dbe1-134a-4f6c-b902-6c2c545f59d9.png)

 but it still shows all the other commodities that are not even present at the station, for example, aluminium and all the other commodities with no supply or demand.
![image](https://user-images.githubusercontent.com/74990247/171163232-5dc8ab43-d2be-423d-82d3-36def221a07d.png)

This station in particular has only a handful of commodities for sale, but even with the buy only filter enabled, the list just goes on and on. 
![image](https://user-images.githubusercontent.com/74990247/171163270-252b1147-f0ca-4c57-9bd4-8fc91a76a19f.png)

now… when comparing stations, this also represents a problem, as it gives false information.

Here we can see how osmium and platinum have a great profit when selling to station B. the problem is, Station A has 0 supply of it.
![image](https://user-images.githubusercontent.com/74990247/171163560-ba8185e8-d776-49da-a2c9-2bf10a6e6efe.png)
And even if I enable the buy only filter, they continue to show up. 
![image](https://user-images.githubusercontent.com/74990247/171163619-f937657c-1543-4efe-a0b5-4253aea79f8f.png)

Other commodities that do have a supply in Station A, will reflect a profit when selling to station B. the problem is Station B does not buy that item at all. 
![image](https://user-images.githubusercontent.com/74990247/171163726-86a8fe79-70d5-4033-aebb-7d56b988e10c.png)
![image](https://user-images.githubusercontent.com/74990247/171163752-882eac8d-5f1d-4c5c-9018-4d604cb3dc06.png)

the "profit from" also have issues as we see here, showing great profits with or without the sell only filter enable… the problem is that any of those commodities are sold in station B
![image](https://user-images.githubusercontent.com/74990247/171163914-ea896f91-f3ee-4a55-bf42-c7a3716a5f6c.png)

even if I were to try and make it work by just organizing by supply, to make sure I’m seeing things that are actually in the station. "profit to" will still show  values. and you could argue that they all have negative values so that’s a giveaway on what is and what is not on the other station. but there are some with negative values that do are in the station. and some other times you want to trade for a negative profit on purpose. and if you are trying to make business using the "profit from" column. well. good luck with that.
![image](https://user-images.githubusercontent.com/74990247/171164075-dfe7e9e0-8a4a-40ad-9e60-1db32ac4ead5.png)
![image](https://user-images.githubusercontent.com/74990247/171164122-33f1bb1f-da0a-4655-a10a-8a970b50c0e9.png)

I have not tested in other markets to confirm or dismiss that they might be cases where the tab shows positive profit and in reality the commodity was not bought in the target station. but in other cases there was also incongruence in the values… like here this hydrogen fuel showing a profit of -4 when in reality it was of 0
![image](https://user-images.githubusercontent.com/74990247/171164309-145788d1-82c5-4cd1-a680-fffa03773fa9.png)
![image](https://user-images.githubusercontent.com/74990247/171164346-76976045-1bb8-4d75-a56d-40ee56869ab3.png)

In conclusion, the market tab can not be trusted… kind of... sort of... well, you need to know what you're doing. 

thanks for coming to my ted talk.

**Comment by Eahlstan** [comment]
Using CAPI the market data only consists of data that the station has a supply or a demand.
Not using the CAPI uses Market.json which has all possible commodities in it, even those with 0 stock and 0 demand.
Those 0/0 items are not filtered.

[Market.zip](https://github.com/EDDiscovery/EDDiscovery/files/8805169/Market.zip)

**Comment by Jugom** [comment]
Do you mean the frontier CAPI from settings? because there I'm logged in
![image](https://user-images.githubusercontent.com/74990247/171177402-d4798775-ba95-4e24-9ac3-41eaa9dab453.png)

**Comment by robbyxp1** [comment]
Thanks for this very valuable input. Your right, it was not working correctly, but we are changing it for 15.1 to fix it and make it better.

closing as acknowledged.
